TheWarWithin/ShamanElemental.lua

```diff
@@ -1239,64 +1231,153 @@ spec:RegisterStateTable( "earth_elemental", setmetatable( { onReset = function(
     end
 } ) )
 
--- The War Within
-spec:RegisterGear( "tww1", 212014, 212012, 212011, 212010, 212009 )
-spec:RegisterAura( "maelstrom_surge", {
-    id = 457727,
-    duration = 5,
-    max_stack = 1
-} )
+spec:RegisterStateTable( "elemental_equilibrium", setmetatable( {
 
-spec:RegisterGear( "tww2", 229260, 229261, 229262, 229263, 229265 )
-spec:RegisterAuras( {
-    -- https://www.wowhead.com/spell=1218612
-    jackpot = {
-        id = 1218612,
-        duration = 8,
-        max_stack = 1
-    },
-} )
+    state = "READY", -- READY, ACTIVE, COOLDOWN
```

Comment:
In order to make time-based forecasting work, this `state` flag should be driven by the remaining time value, similar to how a buff is `up` when `remains` is greater than zero, `remains` is calculated based on `expires - query_time`.  This makes these accurate at any moment in forecasting, instead of just for the instant the value is set.

That also allows us to enable recheck timing in Scripts.lua based on `elemental_equilibrium.ready` flipping when `elemental_equilibrium.cd_remains` (or something) is 0.